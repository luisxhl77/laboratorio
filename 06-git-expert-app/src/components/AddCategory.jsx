import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        //trim() -> elimina espacios en blanco
        //length identifica cuanto strings tiene la variable
        if ( inputValue.trim().length > 0) {

            onNewCategory( inputValue );
            setInputValue('');
        
        }
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text"
                placeholder="buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}