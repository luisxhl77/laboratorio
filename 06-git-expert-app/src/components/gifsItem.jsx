import PropTypes from "prop-types";

export const GifsItem = ({title, url}) => {
  return (
    <div className="card">
        <h4>{title}</h4>
        <img src={url} alt={title} />
    </div>
  )
}

GifsItem.propTypes = {
  
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired

}
