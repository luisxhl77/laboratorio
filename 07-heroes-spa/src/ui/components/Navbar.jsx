import { Link, NavLink } from 'react-router-dom';


export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2">
            
            <Link to="/" className={ ({ isActive }) => `nav-item nav-link ${ isActive ? 'active' : ''} `}>
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink to="/marvel" className={ ({ isActive }) => `nav-item nav-link ${ isActive ? 'active' : ''} `} >
                        Marvel
                    </NavLink>

                    <NavLink to="/dc" className={ ({ isActive }) => `nav-item nav-link ${ isActive ? 'active' : ''} `} >
                        DC
                    </NavLink>
                    
                    <NavLink to="/search" className={ ({ isActive }) => `nav-item nav-link ${ isActive ? 'active' : ''} `} >
                        Search
                    </NavLink>

                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">
                    <span className='nav-item nav-link text-primary'>luis</span>
                    <button className='nav-item nav-link btn'>Logout</button>
                </ul>
            </div>
        </nav>
    )
}