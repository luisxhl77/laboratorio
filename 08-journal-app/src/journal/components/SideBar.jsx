import { TurnedInNot } from "@mui/icons-material"
import { Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"
import { Box } from "@mui/system"

export const SideBar = ({drawerWidth}) => {
    return (
        <Box
            component='nav'
            sx={{ width: {sm: drawerWidth}, flexShrink: {sm: 0} }}
        >
            <Drawer
                variant="permanent"
                open
                sx={{
                    display: { xs: 'block'},
                    '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth},
                }}
            >
                <Toolbar>
                    <Typography variant="h6" noWrap component='div'> Luis Lopez </Typography>
                </Toolbar>
                <Divider/>
                <List>
                    {
                        ['enero','febrero','marzo','abril'].map( text => (
                            <ListItem key={text} disablePadding>
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot/> 
                                        <Grid container>
                                            <ListItemText primary={ text }/>
                                            <ListItemText secondary={ '' }/>
                                        </Grid>
                                    </ListItemIcon>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                </List>
            </Drawer>
        </Box>
    )
}
