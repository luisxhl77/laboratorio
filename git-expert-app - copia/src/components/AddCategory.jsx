import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        //trim() -> elimina espacios en blanco
        //length identifica cuanto strings tiene la variable
        if ( inputValue.trim().length > 0) {
            onNewCategory( inputValue );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
