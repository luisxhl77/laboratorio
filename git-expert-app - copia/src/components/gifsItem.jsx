
export const GifsItem = ({title, url}) => {
  return (
    <div className="card">
        <h4>{title}</h4>
        <img src={url} alt="" />
    </div>
  )
}
