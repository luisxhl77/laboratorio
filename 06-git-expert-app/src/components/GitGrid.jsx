import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifsItem } from './gifsItem';
import PropType from 'prop-types';

export const GitGrid = ({category}) => {
    
    const {images, isLoading} = useFetchGifs( category );  
    //console.log({images, isLoading});

    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading ? (<h2>cargando...</h2>): null
            }
            
            <div className='card-grid'>
                {
                    images.map((image) => (
                        <GifsItem key={image.id} { ...image }/>
                    ))
                }
            </div>
        </>
    )
}

GitGrid.prototype = {
    category: PropType.string.isRequired,
}