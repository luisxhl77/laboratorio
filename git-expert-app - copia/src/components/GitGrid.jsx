import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifsItem } from './gifsItem';

export const GitGrid = ({category}) => {
    
    const {images, isLoading} = useFetchGifs( category );  
    //console.log({images, isLoading});

    return (
        <div className='card-grid'>
            {
                isLoading ? (<h2>cargando...</h2>): null
                //isLoading && (<h2>cargando...</h2>)
            }
            
            {
                images.map((image) => (
                    <GifsItem key={image.id} { ...image }/>
                ))
            }
        </div>
    )
}
