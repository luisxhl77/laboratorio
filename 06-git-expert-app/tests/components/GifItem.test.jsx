import { render, screen } from "@testing-library/react"
import { GifsItem } from "../../src/components/gifsItem"

describe('Prueba de <GifsItem/>', () => {
    const title = "saitama";
    const url = 'https://one-punch.com/saitama.jsg';

    test('debe de hacer match con el snapshop', () => { 

        const { container } = render( <GifsItem title={title} url={url}/> )
        expect( container ).toMatchSnapshot();

    })

    test('debe de mostrar la imagen con el url y el alt indicado', () => {
        
        render( <GifsItem title={title} url={url}/> );
        // expect( screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    
    test('debe de mostrar el titulo en el componente', () => {
        
        render( <GifsItem title={title} url={url}/> );
        expect( screen.getByText(title)).toBeTruthy();

    })
})