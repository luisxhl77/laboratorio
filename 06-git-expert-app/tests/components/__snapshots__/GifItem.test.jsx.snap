// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifsItem/> debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <h4>
      saitama
    </h4>
    <img
      alt="saitama"
      src="https://one-punch.com/saitama.jsg"
    />
  </div>
</div>
`;
